tests/cases/fourslash/codeFixSpellingCaseSensitive4.ts

```diff
@@ -0,0 +1,26 @@
+/// <reference path='fourslash.ts' />
+
+//// declare let a: numbers;
+//// declare let b: Numbers;
+//// declare let c: objects;
+//// declare let d: Objects;
+//// declare let e: RegEx;
+//// namespace yadda {
+////   export type Thing = string;
+//// }
+//// let f: yaddas.Thing;
+
+verify.codeFixAll({
+    fixId: "fixSpelling",
+    fixAllDescription: "Fix all detected spelling errors",
+    newFileContent:
+`declare let a: Number;
```

Comment:
Strange that this doesn't correct to `number` after #39060. Good to look at for future work.